Sofa/GUI/Qt/src/sofa/gui/qt/SofaPluginManager.cpp

```diff
@@ -148,6 +148,16 @@ void SofaPluginManager::addLibrary()
             msg_error("SofaPluginManager") << "plugin should be loaded: " << pluginFile << msgendl;
             return;
         }
+
+        if (plugin->moduleIsInitialized.func && plugin->moduleIsInitialized()
+            || !plugin->moduleIsInitialized.func)
+        {
+            if (const char* moduleName = plugin->getModuleName())
+            {
+                core::ObjectFactory::getInstance()->registerObjectsFromPlugin(moduleName);
```

Comment:
Just for my knowledge, where is this method called normally if not going through the Qt gui ?

Reply:
It should be called whenever you load a plugin. In the future, we could imagine having a closer entanglement between the PluginManager and the ObjectFactory so that you don't have to remember that you have to register objects everytime you load a plugin.